src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py

```diff
@@ -30,17 +31,9 @@ def pa_table_to_pylist(table):
 
 
 def count_path_segments(path):
-    cnt = 0
-    while True:
-        parts = os.path.split(path)
-        if parts[0] == path:
-            break
-        elif parts[1] == path:
-            break
-        else:
-            path = parts[0]
-            cnt += 1
-    return cnt
+    if is_remote_url(path):
+        return path.count("/")
+    return os.path.normpath(path).count(os.sep)
```

Comment:
Yes, we resolve the paths in `data_files.py`, so `normpath` is redundant here.

This should also work, no:
```suggestion
    return path.replace("\\", "/").count("/")
```